onnx/version_converter/adapters/adapter.h

```diff
@@ -0,0 +1,39 @@
+// Interface for Op Version Adapters
+
+#pragma once
+
+#include "onnx/common/ir.h"
+#include "onnx/onnx_pb.h"
+
+namespace ONNX_NAMESPACE { namespace version_conversion {
+
+class Adapter {
+  private:
+    std::string name_;
+    OpSetID initial_version_;
+    OpSetID target_version_;
+
+  public:
+    virtual ~Adapter() noexcept = default;
+
+    explicit Adapter(std::string name, OpSetID initial_version, OpSetID target_version)
```

Comment:
Usually, I would prefer "const std::string& name" + name_(name). You will copy it once anyway. Same strategies apply to OpSetID params.